maven-core/src/main/java/org/apache/maven/execution/MavenExecutionRequest.java

```diff
@@ -277,22 +277,62 @@
 
     MavenExecutionRequest setProfiles( List<Profile> profiles );
 
+    /**
+     * @deprecated Use {@link #getProfileActivation()}.
```

Comment:
We thought about this, but there is not one perfect replacement. For example: `addActiveProfile` could be replaced with either `activateRequiredProfile` or `activateOptionalProfile`. Our (slight) preference was to point the user to the new class instead of having `@deprecated Use this() or that()` everywhere. I don't mind to change it if you prefer that, though.

Reply:
OK, let's leave like that and challenge dev's brain muscles.